src/System.Runtime.InteropServices.RuntimeInformation/src/RuntimeInformation.Windows.cs

```diff
@@ -1,13 +1,77 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Diagnostics;
+
 namespace System.Runtime.InteropServices
 {
-    public static class RuntimeInformation
+    public static partial class RuntimeInformation
     {
+        private static string s_osDebugName = null;
+        private static Architecture? s_osArch = null;
+
         public static bool IsOSPlatform(OSPlatform osPlatform)
         {
             return OSPlatform.Windows == osPlatform;
         }
+
+        public static string GetOSDebugName()
+        {
+            if (null == s_osDebugName)
+            {
+                string type = Interop.mincore.IsWindowsServer() ? "Server" : "Client";
+                string version = "Unknown";
+
+                if (Interop.mincore.IsWindows10OrGreater())
+                    version = "10";
```

Comment:
@stephentoub @weshaggard The api proposed by Stephen and the apis i've used in this PR are not available on onecoreuwp on windows. So this will be rejected by the WACK tool. I am going to use the RTLGetVersion on windows, to just give out whatever windows returns with same behavior as Unix, and on .NET Native, I will just return Windows for now. I have opened up a new issue to track adding or replacing these apis with other windows apis that will be available on all platforms, since it'll take longer to sync up with windows team.